### PR TITLE
fix includes

### PR DESCRIPTION
**Pull request by matijaskala**
needed for musl

### CODE REVIEW

**Comment by poettering** [comment]
I am sorry, but the API we support is glibc's, and that's what we focus on. We are OK with adjusting things where possible and pretty to opt for more generic POSIXy interfaces when both are available, but we explicitly do not support other libcs if they are incompatible with glibc and/or POSIX in this regards. Or to say this differently: instead of working around deviations of API in musl inside of the systemd sources, we ask you to instead correct the musl API, i.e. fix the problem where it originates. This has the benefit that you increase compatibility of musl with any project this way, instead of just adding a specific workaound to one of them.

I hope that makes sense...

That said, some of your changes appear more correct on glibc too

File: src/libsystemd/sd-event/test-event.c [hunk 1]
```diff
@@ -17,6 +17,8 @@
   along with systemd; If not, see <http://www.gnu.org/licenses/>.
 ***/
 
+#include <sys/wait.h>
```
Comment (poettering) [inline src/libsystemd/sd-event/test-event.c:20]: appears to be a muslism, or is there a definition we use that really really needs this header even on POSIX/glibc?

File: src/machine/image-dbus.c [hunk 2]
```diff
@@ -17,6 +17,7 @@
   along with systemd; If not, see <http://www.gnu.org/licenses/>.
 ***/
 
+#include <sys/file.h>
```
Comment (poettering) [inline src/machine/image-dbus.c:20]: same here?

File: src/machine/operation.c [hunk 3]
```diff
@@ -17,6 +17,8 @@
   along with systemd; If not, see <http://www.gnu.org/licenses/>.
 ***/
 
+#include <sys/wait.h>
```
Comment (poettering) [inline src/machine/operation.c:20]: and here?

File: src/nspawn/nspawn-network.c [hunk 4]
```diff
@@ -19,6 +19,7 @@
 
 #include <linux/veth.h>
 #include <net/if.h>
+#include <sys/file.h>
```
Comment (poettering) [inline src/nspawn/nspawn-network.c:22]: and here?

File: src/nspawn/nspawn-stub-pid1.c [hunk 5]
```diff
@@ -18,9 +18,9 @@
 ***/
 
 #include <sys/reboot.h>
-#include <sys/unistd.h>
 #include <sys/wait.h>
 #include <sys/prctl.h>
+#include <unistd.h>
```
Comment (poettering) [inline src/nspawn/nspawn-stub-pid1.c:23]: this one looks like a real bug fix applying to glibc systems too, this one definitely should go in

File: src/sysusers/sysusers.c [hunk 6]
```diff
@@ -19,7 +19,9 @@
 
 #include <getopt.h>
 #include <grp.h>
+#ifdef ENABLE_GSHADOW
 #include <gshadow.h>
+#endif
```
Comment (poettering) [inline src/sysusers/sysusers.c]: why this? gshadow appears to be unconditional part of the glibc API even if not used locally...

File: src/libsystemd/sd-event/test-event.c (hunk 1 above)
Comment (matijaskala) [reply src/libsystemd/sd-event/test-event.c:20]: `WEXITED` - glibc defines it in `stdlib.h` if `_GNU_SOURCE` or `_XOPEN_SOURCE` is defined.

X/OPEN specification says that some macros from `sys/wait.h` should be also defined in `stdlib.h`. However, `WEXITED` is not on that list. glibc seems to define it by accident.

File: src/machine/image-dbus.c (hunk 2 above)
Comment (matijaskala) [reply src/machine/image-dbus.c:20]: This include is for `LOCK_*` macros. Though I'm not sure where should they normally be defined. glibc defines it in `fcntl.h` if `_DEFAULT_SOURCE` or `_GNU_SOURCE` is defined.

**Comment by keszybz** [comment]
@matijaskala based on your comments, it all seems good except for the gshadow change. Can you repush with that last bit dropped?

File: src/libsystemd/sd-event/test-event.c (hunk 1 above)
Comment (poettering) [reply src/libsystemd/sd-event/test-event.c:20]: ok

File: src/machine/image-dbus.c (hunk 2 above)
Comment (poettering) [reply src/machine/image-dbus.c:20]: LOCK_xyz is a BSD API, not a POSIX one, I figure POSIX doesn't matter then... That said, the Linux man pages suggest including sys/file.h for the BSD locking API, hence this appears OK to me.

**Pull request merged**